packages/react-native-executorch/common/rnexecutorch/models/text_to_image/TextToImage.cpp

```diff
@@ -0,0 +1,140 @@
+                         std::shared_ptr<react::CallInvoker> callInvoker)
+    : modelImageSize(imageSize), callInvoker(callInvoker),
+      scheduler(std::make_unique<Scheduler>(schedulerSource, callInvoker)),
+      encoder(std::make_unique<embeddings::TextEmbeddings>(
+          encoderSource, tokenizerSource, callInvoker)),
+      unet(std::make_unique<UNet>(unetSource, imageSize, numChannels,
+                                  callInvoker)),
+      decoder(std::make_unique<Decoder>(decoderSource, imageSize, numChannels,
+                                        callInvoker)) {}
+
+std::shared_ptr<OwningArrayBuffer>
+TextToImage::generate(std::string input, size_t numInferenceSteps) {
+  std::shared_ptr<OwningArrayBuffer> embeddings = encoder->generate(input);
+  std::shared_ptr<OwningArrayBuffer> uncondEmbeddings =
+      encoder->generate(std::string(constants::kBosToken));
+
+  auto *uncondData = reinterpret_cast<float *>(uncondEmbeddings->data());
+  size_t uncondN = uncondEmbeddings->size() / sizeof(float);
+
+  auto *condData = reinterpret_cast<float *>(embeddings->data());
+  size_t condN = embeddings->size() / sizeof(float);
+
+  std::vector<float> embeddingsConcat;
+  embeddingsConcat.reserve(uncondN + condN);
+  embeddingsConcat.insert(embeddingsConcat.end(), uncondData,
+                          uncondData + uncondN);
+  embeddingsConcat.insert(embeddingsConcat.end(), condData, condData + condN);
+
+  // Divide by 8 to account for the 3 down-sampling layers in the VAE model.
+  int32_t latentsWidth = std::floor(modelImageSize / 8);
+  int32_t latentsSize = numChannels * latentsWidth * latentsWidth;
+  std::vector<float> latents(latentsSize);
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::normal_distribution<float> dist(0.0, 1.0);
+  for (auto &val : latents) {
+    val = dist(gen);
+  }
+
+  scheduler->setTimesteps(numInferenceSteps);
+  std::vector<int32_t> timesteps = scheduler->timesteps;
+
+  for (size_t t = 0; t < numInferenceSteps + 1; t++) {
+    log(LOG_LEVEL::Debug, "Step", t, "/", numInferenceSteps);
+    std::vector<float> latentsConcat;
+    latentsConcat.reserve(2 * latentsSize);
+    latentsConcat.insert(latentsConcat.end(), latents.begin(), latents.end());
+    latentsConcat.insert(latentsConcat.end(), latents.begin(), latents.end());
+
+    std::vector<float> noisePred =
+        unet->generate(latentsConcat, timesteps[t], embeddingsConcat);
+
+    int32_t noiseSize = static_cast<int32_t>(noisePred.size() / 2);
+    std::vector<float> noiseUncond(noisePred.begin(),
+                                   noisePred.begin() + noiseSize);
+    std::vector<float> noiseText(noisePred.begin() + noiseSize,
+                                 noisePred.end());
+    std::vector<float> noise(noiseUncond);
+    for (int32_t i = 0; i < noiseSize; i++) {
+      noise[i] =
+          noiseUncond[i] * (1 - guidanceScale) + noiseText[i] * guidanceScale;
+    }
```

Comment:
You use `noiseSize` as a size in for loop and to point iterator in `noiseUncond` and `noisePred` creation. This should be `size_t` then, not signed int type.